### DISCUSSION TITLE
possibly stupid question

### DISCUSSION THREAD

**Discussion by astrograph** [jmcdo29/nest-commander#18] [Q&A] [answered]
so how do you actually call the command from the commandline?

nest BasicCommand
npm run BasicCommand
node BasicCommand
????

**Comment by jmcdo29** [comment, marked as answer]
Best answer is "it depends".

If you're writing an npm package to be a script, you'll need to use the `bin` property of the `package.json` and point it to the `dist/main`. Then when it is installed, the `<feature>` will be a known shell command. This is how my [@ogma/cli](https://github.com/jmcdo29/ogma/tree/main/packages/cli) package works.

If you're making a script that will just be ran only in this repo/project, you can use `ts-node` to run the `main` from `src` if you don't want to compile, or you can compile and use `node dist/main` (or whatever the command file you have is). Then it's just passing the sub-command name (if no default) and the options and arguments.

Let me know if that helps or if I can clarify something more :smile:

**Reply by astrograph** [reply]
thanks for the quick reply.

I tried various things, but in the end I just changed the import `{ something }from @myModule` imports to ` import { something } from './../myModule';`

now I can run my script using

`ts-node src/cli.ts my-command`

**Reply by jmcdo29** [reply]
Ah. If you're using path aliases, you'd probably need to make use of `tsconfig-paths` as well to handle the aliasing. Glad the `ts-node` solution worked for you.
